include/circt/Dialect/Verif/VerifOps.td

```diff
@@ -43,6 +47,43 @@ def CoverOp : AssertLikeOp<"cover"> {
   let summary = "Ensure that a property can hold.";
 }
 
+// Edge behavior enum for always block.  See SV Spec 9.4.2.
+
+/// AtPosEdge triggers on a rise from 0 to 1/X/Z, or X/Z to 1.
+def AtPosEdge: I32EnumAttrCase<"Pos", 0, "posedge">;
+/// AtNegEdge triggers on a drop from 1 to 0/X/Z, or X/Z to 0.
+def AtNegEdge: I32EnumAttrCase<"Neg", 1, "negedge">;
+/// AtEdge is syntactic sugar for AtPosEdge or AtNegEdge.
+def AtEdge   : I32EnumAttrCase<"Both", 2, "edge">;
+
+def ClockEdgeAttr : I32EnumAttr<"ClockEdge", "clock edge",
+                                [AtPosEdge, AtNegEdge, AtEdge]> {
+  let cppNamespace = "circt::verif";
+}
```

Comment:
AFAIK the HW dialect has the same enum (as do LTL and FIRRTL). Is there anything hindering us from just having such an attribute defined once in the HW dialect and using it for all other dialects? That way we wouldn't need to always have those helper functions just converting one enum to the other 1-1.

Reply:
@fabianschuiki recommended redefining the enum to avoid having a dialect dependency (the current pattern being used)

Reply:
Yeah I think the last time we tried to unify this, things didn't go quite as planned because we ended up pulling in other stuff from the dialect and regenerating implementations for other types. I don't know if there's a way to reuse these enum attributes from a different dialect without also generating conflicting declarations/definitions for them.

Reply:
I see. I don't remember when the last time was that we tried to unify this, but there were some quite significant changes related to this in MLIR quite a while ago, so maybe it works now. I'd be quite surprised if sharing attributes isn't possible tbh. But just copy-pasting this single attribute is also a good approach for now, given the uncertainty of the unification approach.

Reply:
I created a "good first issue" for this: https://github.com/llvm/circt/issues/7034